frontend/src/components/VisitRouter/index.js

```diff
@@ -32,7 +32,7 @@ const VisitRouter = observer(() => {
     // kick off api call for program list from Mobx
     participantStore.getPrograms()
     // populate the global visit object if empty (for new visits)
-    if (!Object.keys(participantStore.visit).length) {
+    if (!participantStore.visit.id) {
```

Comment:
checks if an existing database visit is trying to auto populate the form (coming from queue). this is more to the point